lib/block_reservation.rb

```diff
@@ -0,0 +1,20 @@
+module Hotel
+  class BlockReservation < Reservation
+    attr_reader :group_name, :status, :room_qty
+    def initialize(input)
+      super(input)
+      raise ArgumentError, 'Invalid name' if input[:group_name].nil? || input[:group_name].empty?
+      raise ArgumentError, 'Invalid # of rooms' if input[:room_qty].to_i < 2 || input[:room_qty].to_i > 5
+      @included_rooms = input[:included_rooms]
```

Comment:
You should also be able to make reservations from within a block, after the block is "held."